### ISSUE TITLE
only change services on device "reset"

### ISSUE THREAD

**Issue by mmoskal** [microsoft/jacdac#362]
Right now devices can change list of services available at any time.

Proposal:
* whenever the list of services change, the "reset counter" in the announce should go back to "0"
* this can be (but doesn't have to be) done by resetting the device

This will trigger "device disconnected" event in various frameworks, and automatic re-configuration.

As a plus we don't have to track changes to services anymore and we remove an edge case of service list changing (no device does that now). As a minus, there could be an implementation that is less disruptive when service list changes, than when the device resets (but I doubt anyone would want to distinguish these two cases).

**Comment by jamesadevine** [comment]
Sounds good! Is this the behaviour that happens already?

**Comment by mmoskal** [comment]
Yes. We just need to formalize it and remove code handling changes from jd-ts and pxt-jd.

**Comment by pelikhan** [comment]
In MakeCode, it's actually easier to support starting any server at any time.

**Comment by pelikhan** [comment]
This is a problem in MakeCode where users can decide to start a server at any point.

**Comment by pelikhan** [comment]
that works.

**Comment by tballmsft** [comment]
@mmoskal, @pelikhan - close this?

**Comment by pelikhan** [comment]
https://github.com/microsoft/pxt-jacdac/pull/93
